src/extra/gpkg/read.jl

```diff
@@ -0,0 +1,482 @@
+    ORDER BY table_name
+    """)
+  
+  # Use Iterators.drop to get the requested layer
+  rows = Tables.rows(contents_result)
+  layer_iter = Iterators.drop(rows, layer - 1)
+  
+  # Try to get the first row
+  first_row = nothing
+  try
+    first_row = first(layer_iter)
+  catch
+    error("Layer $layer not found in GeoPackage")
+  end
+  
+  table_name = first_row.table_name
+  
+  # Get geometry column information
+  geom_result = DBInterface.execute(db,
+    """
+    SELECT column_name, srs_id
+    FROM gpkg_geometry_columns
+    WHERE table_name = ?
+    """, [table_name])
+  geom_info = first(geom_result)
+  geom_column = geom_info.column_name
+  srs_id = geom_info.srs_id
+  crs_type = get_crs_from_srid(db, srs_id)
+  
+  return table_name, geom_column, crs_type
+end
+
+# This function is no longer needed - table_result is already a valid Tables.jl table
+
+function gpkgreadgeoms(geom_result, geom_column::String, crs_type)
+  geometries = Union{Geometry,Missing,Nothing}[]
+  
+  # Use Tables.rows interface
+  for row in Tables.rows(geom_result)
+    # Use Tables.getcolumn for accessing column values
+    blob = Tables.getcolumn(row, Symbol(geom_column))
+    if ismissing(blob)
+      push!(geometries, missing)
+    elseif isnothing(blob)
+      push!(geometries, nothing)
+    else
+      try
+        geom = parse_gpb(blob, crs_type)
+        push!(geometries, geom)
+      catch
+        # If parsing fails, treat as missing geometry
+        push!(geometries, missing)
+      end
+    end
+  end
+  
+  return geometries
+end
+
+# Get CRS from SRID according to GeoPackage specification
+# Reference: https://www.geopackage.org/spec/#gpkg_spatial_ref_sys_cols
+# SRID possibilities:
+#  0: Undefined geographic coordinate reference system
+# -1: Undefined Cartesian coordinate reference system  
+#  1-32766: Reserved for OGC use (EPSG codes)
+#  32767-65535: Reserved for GeoPackage use
+#  >65535: User-defined coordinate reference systems
+function get_crs_from_srid(db::SQLite.DB, srid::Integer)
+  # Handle special cases per GeoPackage spec
+  if srid == 0
+    # Undefined geographic coordinate reference system - use WGS84 as default
+    return LatLon{WGS84Latest}
+  elseif srid == -1
+    # Undefined Cartesian coordinate reference system
+    return Cartesian{NoDatum}
+  end
+  
+  # Query the database for the organization and organization_coordsys_id
+  srs_result = DBInterface.execute(db,
+    """
+    SELECT organization, organization_coordsys_id
+    FROM gpkg_spatial_ref_sys
+    WHERE srs_id = ?
+    """, [srid])
+  
+  # Try to get the organization info
+  org_info = nothing
+  for row in srs_result
+    org_info = row
+    break
+  end
+  
+  # According to GeoPackage spec, this should not happen in compliant files
+  # but handle gracefully for malformed GeoPackages
+  if isnothing(org_info)
+    @warn "SRID $srid not found in gpkg_spatial_ref_sys table. This violates GeoPackage specification."
+    return Cartesian{NoDatum}
+  end
```

Comment:
Are you encountering this case in practice? Is it coming from GADM or GeoBR tests?

If the file is not compliant, we don't return a warning and `Cartesian`. Please throw an actual error because this is not supposed to occur.